### PR TITLE
feat: added  workflows to delete images and deployment

### PR DESCRIPTION
**Pull request by rajpalc7**
creating github action job to delete images older than 30 days every tuesday, closing PR jobs and stale issues.

### CODE REVIEW

**Comment by WadeBarnes** [comment]
Resolves: https://github.com/bcgov/traction/issues/598

**Comment by rajpalc7** [comment]
Hi @esune , This is ready for testing now. Lets connect whenever you are available.

**Comment by WadeBarnes** [comment]
@rajpalc7, Why the change from `ubuntu-lastest` to `ubuntu-20.04`?

**Comment by rajpalc7** [comment]
Hi @WadeBarnes - Because of stability, security and predictability of 20.04 version . I am thinking latest is working fine today but if there are any security updates or patches tomorrow, our workflow will start failing.

**Comment by WadeBarnes** [comment]
> Hi @WadeBarnes - Because of stability, security and predictability of 20.04 version . I am thinking latest is working fine today but if there are any security updates or patches tomorrow, our workflow will start failing.

In that case wouldn't it be better to use `ubuntu-22.04` since it has a longer support window?

**Comment by rajpalc7** [comment]
ubuntu-22.04 is still at an early access and not as stable as 20.04. I would recommend waiting for few more months before switching our jobs to 22.04 version.

**Comment by WadeBarnes** [comment]
> ubuntu-22.04 is still at an early access and not as stable as 20.04. I would recommend waiting for few more months before switching our jobs to 22.04 version to ensure its stability and reliability.

Could you provide a link to where you're getting this information from?

**Comment by rajpalc7** [comment]
https://www.phoronix.com/news/Ubuntu-22.04-APT-Breaks-Things

**Comment by WadeBarnes** [comment]
> https://www.phoronix.com/news/Ubuntu-22.04-APT-Breaks-Things https://www.youtube.com/watch?v=XotltmuFUyU

The first link is an article dated 14 July 2022, and the video is dated Jun 15, 2022.  Both over a year old.  Is the info still relevant?

**Comment by rajpalc7** [comment]
I read few more articles, they said 22.04 version is better and more stable with latest updates now. I will update our jobs to 22.04 now for longer window support.

File: .github/workflows/delete_images.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Delete old container images
+
+on:
+  schedule:
+    - cron: '0 0 * * 2' # This job runs every Tuesday
+  workflow_dispatch:
+  push:
+    branches:
+      - main
```
[the remark this answers is not in the document]
Comment (rajpalc7) [reply .github/workflows/delete_images.yaml:9]: Hi @esune Thanks for the feedback. So currently this workflow deletes around 100 images in one run, I have set-up this job to trigger every week so that it catches up in few months and comes to 30 days images asap. 

There were 1300 tagged and 50 untagged images before, it has fallen down to 1031 untagged images and 4 untagged images now. Please let me know if you want me increase the limit restrictions so it deletes all images at once.

[the remark this answers is not in the document]
Comment (rajpalc7) [reply .github/workflows/delete_images.yaml]: https://github.com/bcgov/traction/pkgs/container/traction-tenant-ui/versions?page=4

File: .github/workflows/delete_images.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: Delete old container images
+
+on:
+  schedule:
+    - cron: '0 0 * * 2' # This job runs every Tuesday
+  workflow_dispatch:
+  push:
+    branches:
+      - ghcr-clean
```
Comment (esune) [inline .github/workflows/delete_images.yaml:9]: Unsure about why there is a `push` trigger on the `ghcr-clean` branch: as noted previously, I don't think we want to trigger the action on push to any branch, but just run on a schedule and/or manually.

File: .github/workflows/on_pr_closed.yaml [hunk 3]
```diff
@@ -2,19 +2,20 @@ name: Uninstall PR instance
 on:
   pull_request:
     branches:
-      - main
+      - ghcr-clean
```
Comment (esune) [inline .github/workflows/on_pr_closed.yaml:5]: This is incorrect, PRs should be targeting `main` (as it was set to do before).

File: plugins/test.yaml [hunk 4]
```diff
@@ -0,0 +1 @@
+testS
```
Comment (esune) [inline plugins/test.yaml:1]: I assume this is the test file used to trigger the deployment: it should not be part of the PR though.

File: .github/workflows/delete_images.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (esune) [reply .github/workflows/delete_images.yaml]: I took a peek at the registry, a lot of stuff was pruned - excellent. We will need to remove old leftover `pr-xxx` images that have been closed a long time ago to clean it up, we can create a follow-up task once this is completed.

**Review by esune — COMMENTED** [review]
The PR is closed, but the changes have not been merged: are you still working on it @rajpalc7 ?

File: .github/workflows/delete_images.yaml (hunk 2 above)
Comment (rajpalc7) [reply .github/workflows/delete_images.yaml:9]: Hi @esune i am testing stuff so just testing on my branch

File: .github/workflows/on_pr_closed.yaml (hunk 3 above)
Comment (rajpalc7) [reply .github/workflows/on_pr_closed.yaml:5]: Hi @esune i am testing stuff so running on my branch

File: plugins/test.yaml (hunk 4 above)
Comment (rajpalc7) [reply plugins/test.yaml:1]: Hi @esune i am testing stuff to trigger the workflow

**Comment by rajpalc7** [comment]
Hi @esune, I was able to re-test all the 3 jobs successfully. I can demo it tomorrow and we can go there.
